### PR TITLE
Refactor Input parameters

### PR DESCRIPTION
**Pull request by alejoberardino**
Input parameters in their current state do not support complex validations, and it’s complicated to enforce consistency between the transport/endpoint generics and the params themselves. This refactor changes them to a format similar to the AdapterConfig that allows for the same rough details, while making it easier to code and enforce types.

### CODE REVIEW

File: src/validation/input-params.ts [hunk 1]
```diff
@@ -1,28 +1,475 @@
...
+type InputParameterOptionConstraints =
+  | {
+      // Only allow options for string params
+      type: 'string'
+      options?: string[] | readonly string[] // Enumerated options, ex. ['ADA', 'BTC', 'ETH']
+    }
+  | {
+      type: Exclude<ParameterType, 'string'>
+      options?: never
+    }
```
Comment (amit-momin) [inline src/validation/input-params.ts:83]: This was probably already the restriction but feel like there could be a use case for number options. Like if we wanted to restrict PoR adapter `chainId` params to specific ones for ethereum. Or if a data provider has a parameter to specify specific windows that are numbers.

Comment (alejoberardino) [reply src/validation/input-params.ts]: Fair enough, `chainId` is the counterargument to not allowing it. Changes pushed

File: src/validation/input-params.ts [hunk 2]
```diff
@@ -1,28 +1,478 @@
...
+      options?: string[] | readonly string[]
+    }
+  | {
+      type: 'number'
+      options?: number[] | readonly number[]
+    }
+  | {
+      type: Exclude<ParameterType, 'string'>
+      options?: never
+    }
```
Comment (amit-momin) [inline src/validation/input-params.ts:86]: Wouldn't this need to be this now?
```suggestion
  | {
      type: Exclude<ParameterType, 'string' | 'number'>
      options?: never
    }
```

Comment (alejoberardino) [reply src/validation/input-params.ts]: Fixed

**Review by amit-momin — APPROVED** [review]

**Pull request merged**